server/src-lib/Hasura/GraphQL/Execute.hs

```diff
@@ -70,7 +70,7 @@ import qualified Hasura.Logging                         as L
 -- The 'a' is parameterised so this AST can represent
 -- intermediate passes
 data GQFieldPartialPlan
-  = GQFieldPartialHasura !(GCtx, VQ.Field)
+  = GQFieldPartialHasura !VQ.Field
   | GQFieldPartialRemote !RemoteSchemaInfo !VQ.Field
```

Comment:
TBH, `RemoteSchemaInfo` is here only because `GCtx` was present for `GQFieldPartialHasura`. Now that, this is gone we can resolve RemoSchemaInfo in `getResolvedExecPlan` function only.

Reply:
Hm, don't we need RemoteSchemaInfo in `execRemoteGQ ` eventually? I'm not sure I follow how this would work

Reply:
This is the representation partial plan and we can perhaps resolve `RemoteSchemaInfo` in the "resolved" plan i.e. in `getResolvedExecPlan` ?